### ISSUE TITLE
Q: how different SuperPosition algorithms compare?

### ISSUE THREAD

**Issue by wojdyr** [biojava/biojava#892]
My question is more about algorithms than about BioJava, but since BioJava has three superposition methods implemented I thought I'd try asking here.
What are good and bad sides of QCP, SVD(Kabsch) and Quat(ernion based algorithm)?
I'm reading a paper about QCP, but it focuses on the good sides only. I wonder: do other methods have advantages over QCP?
Thanks,
Marcin

**Comment by lafita** [comment]
Hi Marcin,

We did some comparison between the different methods in BioJava some time ago, as they are heavily used by other algorithms. The main advantage of QCP is that the RMSD can be estimated without the need to calculate the full point superpositon. In principle this allows to estimate the RMSD and superposition matrix in two sequential steps, instead of a single one, and is faster if only the RMSD is required. Apart from that the three algorithms should have the same time complexity, as far as I am aware.

We have a few tests that compare the different superposition implementations on sets of fake points here:
https://github.com/biojava/biojava/tree/master/biojava-structure/src/test/java/org/biojava/nbio/structure/geometry

Performance-wise we found that SVD scales the worst and Quat was the best choice for proteins (50-500 CA atoms), but QCP was close. These are some running times from the test as an example:
```
50 points: SVD time 184 us, Quat time: 195 us, QCP time: 185 us
500 points: SVD time 750 us, Quat time: 217 us, QCP time: 373 us
5000 points: SVD time 4308 us, Quat time: 1783 us, QCP time: 1933 us
```

The QCP test also showed that computing the RMSD only is faster that doing the full superposition.
```
Time for RMSD: 1101 us, superposition after RMSD: 85 us, and superposition from scratch: 1505 us
```

Hope this helps!
Aleix
